### PR TITLE
[Tune] Raise error when PBT is used with search algorithm

### PR DESCRIPTION
**Pull request by amogkam**
We should raise an error when PBT or PB2 are used with a search algorithm to prevent usages like in https://github.com/huggingface/transformers/issues/10247

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/tune/tune.py [hunk 1]
```diff
@@ -364,6 +365,12 @@ def run(
             "The `mode` parameter passed to `tune.run()` has to be one of "
             "['min', 'max']")
 
+    if scheduler is not None and isinstance(scheduler,
+                                            PopulationBasedTraining) and \
+            search_alg is not None:
+        raise ValueError("Search algorithms cannot be used with PBT or PB2 "
+                         "schedulers. Please remove {}.".format(search_alg))
+
```
Comment (richardliaw) [inline python/ray/tune/tune.py:373]: Can we instead raise this in pbt.py, like in `on_trial_add`?

Comment (amogkam) [reply python/ray/tune/tune.py:373]: My concern with that is that won't that actually attempt to start training, and then raise errors in each remote training processes? That could possibly result in confusing errors.

This is a usage error with `tune.run` so I thought we can preemptively raise an error on the driver before training even begins.

Comment (richardliaw) [reply python/ray/tune/tune.py:373]: No, on_trial_add is called before any remote processes start, so it should be safe to raise an error there

Comment (amogkam) [reply python/ray/tune/tune.py]: Ah ok, got it. I pushed a change for it.

File: python/ray/tune/schedulers/pbt.py [hunk 2]
```diff
@@ -319,6 +319,12 @@ def set_search_properties(self, metric: Optional[str],
 
     def on_trial_add(self, trial_runner: "trial_runner.TrialRunner",
                      trial: Trial):
+        if trial_runner.search_alg is not None:
```
Comment (richardliaw) [inline python/ray/tune/schedulers/pbt.py:322]: it can also be BasicVariantGenerator right?

Comment (amogkam) [reply python/ray/tune/schedulers/pbt.py]: Good point, updated.

**Review by richardliaw — APPROVED** [review]
LGTM! Ping when tests pass

**Comment by amogkam** [comment]
@richardliaw can you merge please, thanks!

**Pull request merged**